dataprofiler/data_readers/graph_data.py

```diff
@@ -0,0 +1,159 @@
+import csv
+
+import networkx as nx
+from numpy import source
+
+from .base_data import BaseData
+from .csv_data import CSVData
+from .filepath_or_buffer import FileOrBufferHandler
+
+class GraphData(BaseData):
+        
+    def __init__(self, input_file_path=None, data=None, options=None):
+        BaseData.__init__(self, input_file_path, data, options)
+        if options is None:
+            options = dict()
+
+        return self._load_data()
+
+
+    @classmethod
+    def _find_target_string_in_column(self, column_names, keyword_list):
+        '''
+        Find whether one of the columns names contains a keyword that could refer to a target node column
+        '''
+        column_name_symbols = ['_', '.', '-']
+        has_target = False
+        target_index = -1
+        
+        # iterate through columns, keywords, and delimiter name symbols to see if any permutation is contained in column names
+        for column in range(0, len(column_names)):
+            for keyword in keyword_list:
+                for symbol in column_name_symbols:
+                    
+                    append_start_word = symbol + keyword
+                    append_end_word = keyword + symbol
+
+                    if append_start_word in column_names[column] or append_end_word in column_names[column]:
+                        target_index = column
+                        has_target = True
+                        break
+            if has_target:
+                break
+        
+        return target_index
+
+
+    @classmethod
+    def csv_column_names(cls, file_path, options):
+        '''
+        fetches a list of column names from the csv file
+        '''
+        column_names = []
+
+        with FileOrBufferHandler(file_path) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter = options.get("delimiter", ","))
+            
+            # fetch only column names
+            for row in csv_reader:
+                column_names.append(row)
+                break
+        column_names = column_names[0]
```

Comment:
right now this presumes the first row is the header. Some CSVs have comments at the top. We cause use the info from `CSVData.is_match` and options to determine this. (will also need a test, e.g. single graph data file which has comments before the column names start).
to make this more generic we could do:
```python
@classmethod
def csv_column_names(cls, file_path, delimiter, quotechar, header): 
    column_names = []
    if header is None:
        return column_names

    for _ in range(header + 1):
        row = next(csv_reader)
    .
    .
    .
```
csv.reader can be updated to handle quotechar too:
```python
csv_reader = csv.reader(csv_file, delimiter=delimiter, quotechar=quotechar)
```